platforms/core-configuration/configuration-cache/src/integTest/groovy/org/gradle/internal/cc/impl/ConfigurationCacheBuildServiceIntegrationTest.groovy

```diff
@@ -647,10 +651,10 @@ class ConfigurationCacheBuildServiceIntegrationTest extends AbstractConfiguratio
         configurationCache.assertStateLoaded()
 
         where:
-        injectionAnnotation                    | propertyType           | taskConfiguration                                       | valueGetter
-        "${ServiceReference.name}('constant')" | "ConstantBuildService" | ""                                                      | "value.get().value"
-        Internal.name                          | "ConstantBuildService" | "value = serviceProvider; usesService(serviceProvider)" | "value.get().value"
-        Input.name                             | "String"               | "value = serviceProvider.map { it.value }"              | "value.get()"
```

Comment:
❔ Was it possible to call the provider indirectly without declaring service usage 😮 ?

Reply:
we have `@Input Property<String>` with `serviceProvider.map { it.value }` as a value, and no `usesService`. So there should be a warning. But it looks like we're not enabling the `STABLE_CONFIGURATION_CACHE` for this test or it happened too late to have an effect.

Reply:
Ah, I see, thanks